internal/app/machined/internal/phase/services/start_services.go

```diff
@@ -81,3 +85,20 @@ func (task *StartServices) loadKubernetesServices(r runtime.Runtime) {
 		)
 	}
 }
+
+func (task *StartServices) wait(r runtime.Runtime) (err error) {
+	svcs := system.Services(r.Config()).List()
```

Comment:
waiting for all the services to start might break some stuff a bit for those 5 minutes. if some service doesn't start for any reason, boot sequence will never finish, so `machined` will never get into the state when it accepts other sequences (reboot for example)

not a big deal I guess, but I thought it might be important

Reply:
Yeah, but unfortunately this is the only surefire way to know that the upgrade was successful. It is a tradeoff :/

Reply:
other option I thought about is not that clean and breaks out of sequencer, but basically spin up goroutine here, wait for services to be healthy, and only after that do upgrade cleanup

Reply:
Yeah that is a possibility. I feel like we should stick with the sequencer paradigm as close as possible, and in the next version of the sequencer we can address the limitations in the current implementation, then fix this (and a number of other things). And, actually, today the `post startup tasks` continuously tries to mark control plane nodes with the master label, effectively blocking APIs, even today without this change. So nothing really changes from how things work today.